runtime/backtrace.c

```diff
@@ -156,9 +158,15 @@ CAMLprim value caml_get_exception_raw_backtrace(value unit)
     memcpy(saved_backtrace_buffer, Caml_state->backtrace_buffer,
            saved_backtrace_pos * sizeof(backtrace_slot));
 
-    res = caml_alloc(saved_backtrace_pos, 0);
-    for (i = 0; i < saved_backtrace_pos; i++) {
-      Field(res, i) = Val_backtrace_slot(saved_backtrace_buffer[i]);
+    res = caml_alloc_shr_no_track_noexc(saved_backtrace_pos, 0);
+
+    if (res) {
```

Comment:
I think this would be more readable with the exceptional case first: `if (res == 0) ...`.